Mes 8870 journal refresh bug

gittins7 opened the pull request:
## Description

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)

gittins7 committed:
WIP: journal refresh fix

gittins7 committed:
WIP: Attempt to fix journal refresh bug on initial app launch

gittins7 committed:
Removed any no longer necessary console.logs, taps & added an action dispatch to dashboard.ts to fix journal rehydration issue

gittins7 committed:
removed old console logs and reintroduced accidentally removed tap

gittins7 committed:
Refactored dispatching of action to only happen on navigation to journal from dashboard instead of on dashboard entry

gittins7 committed:
Unit tests

matthew2564 commented on src/app/providers/search/search.ts:
```diff
@@ -39,20 +39,11 @@ export class SearchProvider {
   }
 
   advancedSearch(advancedSearchParams: AdvancedSearchParams): Observable<SearchResultTestSchema[]> {
+    const searchParams = JSON.parse(JSON.stringify(advancedSearchParams));
```
This will break advanced search, `dtcCode` will now not being sent and `costCode` will instead.

Also, we have now lost the ability to `encodeURIComponent`.

matthew2564 commented on src/app/providers/search/__tests__/search.spec.ts:
```diff
@@ -55,16 +55,16 @@ describe('SearchProvider', () => {
       httpMock.expectOne(
         // eslint-disable-next-line max-len
-        'https://www.example.com/api/v1/test-result?startDate=12-12-12&endDate=12-12-12&staffNumber=12345&dtcCode=abc&excludeAutoSavedTests=true&category=A&activityCode=1&rekey=true&passCertificateNumber=A1',
+        'https://www.example.com/api/v1/test-result?startDate=12-12-12&endDate=12-12-12&staffNumber=12345&costCode=abc&excludeAutoSavedTests=true&activityCode=1&rekey=true&category=A&passCertificateNumber=A1',
```
This has changed but shouldn't have when provider comment is addressed.

matthew2564 commented on src/app/pages/dashboard/dashboard.page.ts:
```diff
@@ -108,13 +108,21 @@ export class DashboardPage extends BasePageComponent {
   }
 
+  async navigate(pageName: string) {
+    if (pageName === 'JournalPage') {
```
Can we use the `JOURNAL_PAGE` constant here.

matthew2564 commented:
As mentioned before, we need to be smarter about when we are calling `this.store$.dispatch(LoadCompletedTests(true));` inside Journal Page. 

We need to guard the call to not run when navigating from Dashboard.

matthew2564 commented on src/app/pages/dashboard/components/go-to-journal-card/go-to-journal-card.ts:
```diff
@@ -9,10 +9,11 @@ import { JOURNAL_PAGE } from '../../../page-names.constants';
-
+  @Output()
+  navigateClicked = new EventEmitter<string>();
   constructor(private router: Router) { }
```
This should now be able to be removed, as well as via the spec if it is injected

gittins7 committed:
WIP: advanced search params change rewrite to to PR comment on MES-8870

gittins7 committed:
WIP: passCertNumber change

gittins7 committed:
PR comment amendments

gittins7 committed:
Unit test

gittins7 committed:
Removed unused import

matthew2564 commented on src/app/app.component.ts:
```diff
@@ -110,6 +110,7 @@ export class AppComponent extends LogoutBasePageComponent implements OnInit {
       this.store$.dispatch(LoadAppVersion());
       await this.configureStatusBar();
       this.configureLocale();
+      alert('hello');
```
Please remove

matthew2564 commented on src/app/pages/dashboard/dashboard.page.ts:
```diff
@@ -173,6 +175,13 @@ export class DashboardPage extends BasePageComponent {
 
+  async navigate(pageName: string) {
+    if (pageName === JOURNAL_PAGE) {
+      this.store$.dispatch(journalActions.LoadCompletedTests(true));
```
As this was purely a dev issue, thought we said we would forget about this.

matthew2564 commented on src/app/providers/search/search.ts:
```diff
@@ -39,21 +41,10 @@ export class SearchProvider {
   }
 
   advancedSearch(advancedSearchParams: AdvancedSearchParams): Observable<SearchResultTestSchema[]> {
+    const params = JSON.parse(JSON.stringify(stripNullishValues(advancedSearchParams)));
```
This doesnt need to be wrapped with parse/stringify any longer

gittins7 committed:
Removed all code relating to initial navigation change as it's only an issue in dev
Removed alert in app.component.ts
Removed JSON.parse() & JSON.Stringify() implementation as it's no longer needed
